service/controller/v13/resource/instance/create_terminate_old_workers.go

```diff
@@ -0,0 +1,73 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) terminateOldWorkersTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding all worker VMSS instances")
+	var allWorkerInstances []compute.VirtualMachineScaleSetVM
+	{
+		allWorkerInstances, err = r.allInstances(ctx, customObject, key.WorkerVMSSName)
+		if IsScaleSetNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the scale set '%s'", key.WorkerVMSSName(customObject)))
```

Comment:
Not sure whe scenario where this would happen, but would it make sense to move forward to the next state instead? I'm thinking that if we hit this scenario, the operator will get stuck on this state. We could print a warning but keep moving forward if the VMSS is not there? Just thinking out loud here.

Reply:
Would it even make sense to go to beginning?

Reply:
You mean `DeploymentUninitialized`?

Reply:
Yeah

Reply:
That could be a good idea yeah.